src/pages/docs/[[...filePath]].tsx

```diff
@@ -0,0 +1,173 @@
+};
+
+const indexLink = (
+  <div style={{ margin: "32px auto", maxWidth: "1000px" }}>
+    <Link href="/docs">
+      <Typography>Back to documentation index</Typography>
+    </Link>
+  </div>
+);
+
+const homeLink = (
+  <div style={{ margin: "32px auto", maxWidth: "1000px" }}>
+    <Link href="/">
+      <Typography>Back to home</Typography>
+    </Link>
+  </div>
+);
+
+const header = <Typography variant="h1">
+  <Link href="/">
+    <img src="/img/vns-logo-96.png" style={{ marginRight: "32px" }} />
+  </Link>
+  VN Live Documentation
+</Typography>;
+
+interface PageArguments {
+  pages: Array<string>,
+  filePath: string,
+  source: MdxRemote.Source
+}
+export default function DocPage({ pages, filePath, source }: PageArguments) {
+  if (source) { // It's a file, not a folder
+    const content = hydrate(source, {
+      components,
+    });
+    return (
+      <div className="MDXProvider root">
+        {header}
+        {indexLink}
+        {content}
+        {indexLink}
+        <style jsx>{`
+        .MDXProvider.root {
+          margin: 32px auto;
+          max-width: 1000px;
+        }
+        `}</style>
+      </div>
+    );
+  }
+  else { // It's a folder
+    return (
+      <div style={{ margin: "32px auto", maxWidth: "1000px" }}>
+        {header}
+        <Typography style={{ textTransform: "capitalize" }} variant="h2"> {filePath} </Typography> {/* Print the subfolders we're in */}
+        <List>
+          {
+            pages.map((pageName) => (
+              <NextLink key={pageName} href={`/docs/${filePath}${pageName}`}>
+                <ListItem button key={pageName}>
+                  <Typography style={{ textTransform: "capitalize" }}>
+                    {
+                      pageName.replace(
+                        /-/g,
+                        " "
+                      ) /* we don't use the front matter of the file at this point to simplify loading, so we have to cleanup the name manually */
+                    }
+                  </Typography>
+                </ListItem>
+              </NextLink>
+            ))
+          }
+          <hr></hr>
+          {filePath === '' /* Back home if we're in /docs, back to /docs if we're in a subfolder */
+            ? homeLink
+            : indexLink
+          }
+        </List>
+      </div>
+    )
+  }
+}
+
+interface PathsProps {
+  params: { filePath: Array<string> }
+}
+export async function getStaticPaths() {
+  const docsDir = path.resolve("./src/content/docs"); // relative to the project root
+  const files = await getMdxPaths(docsDir);
+  const pageNames = files.map((f) =>
+    f.params.fileName.join('/')
+  );
+  // paths is the file without the extension, shaped as [{ params: { filePath: [ 'subdirectory', 'file' ] } } ]
+  let paths: Array<PathsProps> = [{ params: { filePath: [''] } }];
+  for (let name of pageNames) {
+    const splittedName = name.split('/');
+    for (let iNbSplit = 0; iNbSplit < splittedName.length; iNbSplit++) {
```

Comment:
you can do a splittedName.forEach(namePart => {...}) instead, for loops are not very readable